include/mapbox/geometry/wagyu/almost_equal.hpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2005, Google Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
```

Comment:
I think this extra LICENSE should be referenced in the main (https://github.com/mapbox/wagyu/blob/master/LICENSE) license file for clarity.

Reply:
@Algunenano  great point! Any chance you could help by providing an example of how to do this for multiple licenses?

Reply:
To introduce Wagyu in Postgis I added a [LICENSE.README](https://github.com/postgis/postgis/blob/master/deps/wagyu/LICENSE.README) detailing what license affects what files and multiple licenses (LICENSE.geometry and LICENSE.wagyu). I'm not sure if there's a better way but that seemed like a valid option.

Reply:
@Algunenano after looking into this some I realized we already had some complications in the LICENSE file as is with the Angus clipper stuff as well. Could you verify that the current changes work?

Reply:
I think that is ok, but keep in mind that now the library requires adding google's copyright also on binary distributions. This can be somewhat a pain for projects downstream, for example, node-mapnik will now be forced to include this in its LICENSE so it appears in the binary distribution (the files uploaded to s3). It's easy to miss it.

Reply:
I spent quite a bit of time looking over how to possibly do this with out that specific code -- to perhaps have a different license to not include that specific code. I haven't found any other great solution yet in any code that has no restrictions but was also easy to put into the library.

I suppose I could find and try to port Boost's implementation into this library -- https://www.boost.org/doc/libs/1_72_0/libs/math/doc/html/math_toolkit/float_comparison.html

I could also perhaps include boost as a dependency, but that seems like a pain too. I did some extensive testing with this new change and it seems to currently work in most situations though so I am hesitant to change it.

Reply:
The blogpost refenced in the gtest code (https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/) is published without license (`No restrictions. Also no warranty. Use it for what you will. I’d recommend a link back to the article just so people will understand how it works.`).

Would it be ok to adapt it for this use case (double and int64_t), without all the extra templates that aren't necessary here? I'm thinking something along these lines (the bit mask might need some tweaking):
```c
/* See
https://randomascii.wordpress.com/2012/01/11/tricks-with-the-floating-point-format/
for the potential portability problems with the union and bit-fields below.
*/
#include <stdint.h> // For int64_t, etc.
 
union Double_t
{
    Double_t(double num = 0.0f) : f(num) {}
    // Portable extraction of components.
    bool Negative() const { return i < 0; }
    int64_t RawMantissa() const { return i & ((1 << 52) - 1); }
    int64_t RawExponent() const { return (i >> 52) & 0xFF; }
 
    int64_t i;
    double f;
};
 
bool AlmostEqualUlps(double A, double B, int maxUlpsDiff)
{
    Double_t uA(A);
    Double_t uB(B);
 
    // Different signs means they do not match.
    if (uA.Negative() != uB.Negative())
    {
        // Check for equality to make sure +0==-0
        if (A == B)
            return true;
        return false;
    }
 
    // Find the difference in ULPs.
    int ulpsDiff = abs(uA.i - uB.i);
    if (ulpsDiff <= maxUlpsDiff)
        return true;
 
    return false;
}
```

Reply:
After studying this a bit more I think I like the google test implementation better still, going to just go with this for now.  Also talked to some others and apparently this is just BSD License so I think we just need to include that in the binary distributions.